File: lesson_11/arrays_java/arrays_app/src/main/java/com/codedifferently/lesson11/Lesson11.java
package com.codedifferently.lesson11;

import java.util.ArrayList;
import java.util.List;

public class Lesson11 {

  /**
   * Provide the solution to LeetCode 1929 here:
   * https://leetcode.com/problems/concatenation-of-array
   */
  public int[] getConcatenation(int[] nums) {
    int n = nums.length;
    int[] result = new int[n * 2];
    for (int i = 0; i < n; i++) {
      result[i] = nums[i];
      result[i + n] = nums[i];
    }
    return result;
  }

  /**
   * Provide the solution to LeetCode 2942 here:
   * https://leetcode.com/problems/find-words-containing-character/
   */
  public List<Integer> findWordsContaining(String[] words, char x) {
    int n = words.length;
    List<Integer> result = new ArrayList<>();
    for (int i = 0; i < n; i++) {
      if (words[i].indexOf(x) != -1) {
        result.add(i);
      }
    }
    return result;
  }
}

File: lesson_11/arrays_ts/src/lesson11.ts
/**
 * Provide the solution to LeetCode 1929 here:
 * https://leetcode.com/problems/concatenation-of-array
 */
export function getConcatenation(nums: number[]): number[] {
  const n = nums.length;
  const result = new Array(n * 2);
  for (let i = 0; i < n; i++) {
    result[i] = nums[i];
    result[i + n] = nums[i];
  }
  return result;
}

/**
 * Provide the solution to LeetCode 2942 here:
 * https://leetcode.com/problems/find-words-containing-character/
 */
export function findWordsContaining(words: string[], x: string): number[] {
  const result: number[] = [];
  const n = words.length;
  for (let i = 0; i < n; i++) {
    if (words[i].indexOf(x) !== -1) {
      result.push(i);
    }
  }
  return result;
}
